src/common/pathfinding.ts

```diff
@@ -1,14 +1,28 @@
 import * as jkstra from 'jkstra';
 import { HexBoardGraph } from './graph';
+import { Coordinates } from './utils';
 
 export function doesPathExist(
   hexBoardGraph: HexBoardGraph,
   startNodeId: string,
   endNodeId: string,
-): boolean {
+): { hasWon: boolean; winningPath: Coordinates[] } {
   const dijkstra = new jkstra.algos.Dijkstra(hexBoardGraph.graph);
-  return !!dijkstra.shortestPath(
+  const shortestPath = dijkstra.shortestPath(
     hexBoardGraph.vertexMap.get(startNodeId),
     hexBoardGraph.vertexMap.get(endNodeId),
   );
+  if (!!shortestPath) {
```

Comment:
`!!` is useful for ensuring you return a boolean, but here I think `if (shortestPath)` is enough